From f4e30b839a02c2518688306bc213278f75d19a95 Mon Sep 17 00:00:00 2001
From: Kien Nguyen <kien.n.quang@gmail.com>
Date: Wed, 6 Apr 2022 09:21:32 +0900
Subject: [PATCH] Make file-name-split returns driver name as well in Windows

* lisp/files.el: make file-name-split returns driver name as well in Windows (bug#54721)

---
 lisp/files.el | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lisp/files.el b/lisp/files.el
index a0bc5bf262..c198f18723 100644
--- a/lisp/files.el
+++ b/lisp/files.el
@@ -5091,7 +5091,7 @@ file-name-split
         ;; If there's nothing left to peel off, we're at the root and
         ;; we can stop.
         (when (and dir (equal dir filename))
-          (push "" components)
+          (push (if (string-empty-p dir) "" (substring dir 0 -1)) components)
           (setq filename nil))))
     components))
 
-- 
2.35.1.windows.2

